contentcuration/contentcuration/assets/shared/views/ThumbnailUpload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div
+    :style="highlight && !readonly ? { borderColor: 'purple' } : {}"
+    @dragenter.prevent="enter"
+    @dragover.prevent="over"
+    @dragleave.prevent="leave"
+    @drop.prevent="drop"
+  >
+    <template
+      v-for="file in files"
+    >
+      <VProgressCircular
+        v-if="file.progress < 100"
+        :key="file.id"
+        :size="50"
+        :width="15"
+        :value="file.progress"
+      >
+        {{ file.progress }}
+      </VProgressCircular>
+    </template>
+    <input
+      v-if="!readonly"
+      id="fileElem"
+      ref="fileUpload"
+      style="display: none;"
+      type="file"
+      accept="image/*"
+      @change="handleFiles($event.target.files)"
+    >
+    <label for="fileElem">
+      <VImg v-if="file" contain :src="file" />
+      <VImg v-else contain :src="require('../images/kolibri-logo.svg')" />
+      <VBtn v-if="!readonly" @click="$refs.fileUpload.click()">
+        {{ $tr('fileUpload') }}
+      </VBtn>
+    </label>
+  </div>
+</template>
+
+<script>
+
+  import client from 'shared/client';
+
+  export default {
+    name: 'ThumbnailUpload',
+    $trs: {
+      fileUpload: 'Upload file',
+    },
+    props: {
+      value: {
+        type: Object,
+        default() {
+          return {};
+        },
+      },
+      readonly: {
```

Comment:
Use this to allow us to set the input to readonly in cases where someone is viewing a thumbnail for something they cannot edit.